External modules3

kofifus opened the pull request:
This PR makes cornerstoneWADOImageLoader ready for direct browser loading

- add dicomParser to the external exported from externalModules.js

- make 'export' the default export from externalModules.js (as is in cornerstoneTools)

- make window.cornerstone and window.dicomParser the defaults if available

- removed externalModulesES6modules.js as it is no longer needed

- removed `export { internal } from './internal/index.js';` from imageLoader/index.js as it not used anywhere

- removed the need to supply the cornerstone instance to the registerLoaders() and the register function in wadors/register.js and wadouri/register.js

- changed tests according to above

- update package.json to latest version of packages as used in the other cornerstone packages

swederik commented on src/externalModules.js:
```diff
@@ -1,18 +1,25 @@
   set cornerstone (cs) {
-    cornerstone = cs;
-
-    registerLoaders(cornerstone);
```
Don't we still want this to occur?

kofifus replied:
solved see comment below

swederik commented:
Hmm so this would require a major version bump I guess

dannyrb commented on src/imageLoader/wadouri/dataSetCacheManager.js:
```diff
@@ -28,8 +27,7 @@ function get (uri) {
 function load (uri, loadRequest = xhrRequest, imageId) {
-  const cornerstone = external.cornerstone;
-
+  const { cornerstone, dicomParser } = external;
```
Ooh. I actually think this is cleaner than what we had, but why do it here instead of at the top with the rest of the imports?

dannyrb commented:
> This PR makes cornerstoneWADOImageLoader ready for direct browser loading

Meaning you can include it from a packaged source, rather than as part of a build step?

> Code using cornerstoneWADOImageLoader should now call registerLoaders() before starting like this:
```
import * as cornerstoneWADOImageLoader from 'cornerstone/cornerstoneWADOImageLoader/src/imageLoader/index.js';
cornerstoneWADOImageLoader.registerLoaders();
```

I feel like there has to be a cleaner way. What are we trying to accomplish here? Ease of setup from a packaged source OR from `npm` as part of a build step? The code may accomplish that, but I'm having a hard time Grok'ing it to verify. I imagine the end goal would be:

_From Packaged Source_

- Add source to page
    - Image loader + config is added to `window.cornerstoneLoaders` array
- Add cornerstone source to page
    - "Start Cornerstone"
    - All image loaders in `window.cornerstoneLoaders` array are iterated over and registered

_With NPM / Build Step_

- Same business as usual


I'm spit-balling. I may be missing what you're trying to accomplish entirely. For breaking / API changes like this, it's often best to create an issue to propose changes so we can discuss implementations beforehand.

kofifus commented:
wow I just updated my chrome and it now seems to handle circular import dependencies just fine ! this must have been a very recent change which is great cause the PR is no longer a breaking change!

I updated the PR and the tests accordingly. Tests pass fine and I can load the code directly in Chrome without a problem ...

I also update package.json to latest version of packages as used in the other cornerstone packagers.

Thanks!

kofifus commented:
Hi can this be merged or is there any help I can give to towards having it merged ?

We have a few developers working with this for a while and all is well ..
